api.ts

```diff
@@ -99,6 +99,21 @@ export class UserVisibleError extends Error {
   }
 }
 
+/**
+ * An error that occurs when validating parameters.
+ */
+export class ParameterValidationError extends UserVisibleError {
+  /**
+   * The parameters that were invalid, alongside a message describing the error for the parameter.
+   */
+  invalidParameters: Record<string, string>;
```

Comment:
Nit, I think it's a little more flexible and consistent to have this be an array of error objects. That allows for enriching the error data, for example by having enumerated error codes/types. And it's also kind of an established pattern, e.g. zod does this, and our own upload_validation.ts ultimately gives an array of errors, with each having a message and a path. So something like:

```ts
readonly errors: ParameterError[];
...
export interface ParameterError {
  message: string;
  parameterName: string;
}
```